src/main/java/TaskList.java

```diff
@@ -0,0 +1,47 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    protected int curr = 0, end = 0, len = 0;
```

Comment:
Naming of these variables seems abit ambiguous, would be better if they reference what curr, end, and len are referring to. (which in this case would be the indexes of tasks)